### PR TITLE
handle expanding reduce

### PR DESCRIPTION
**Pull request by havardpe**
more testing of corner cases

@arnej27959 please review

### CODE REVIEW

File: eval/src/vespa/eval/instruction/universal_dot_product.cpp [hunk 1]
```diff
@@ -56,42 +56,60 @@ void my_universal_dot_product_op(InterpretedFunction::State &state, uint64_t par
...
+        std::fill(empty.begin(), empty.end(), OCT{});
+    }
     state.pop_pop_push(result);
 }
 
+template <typename OCT>
+const Value &create_empty_result(const UniversalDotProductParam &param, Stash &stash) {
+    if (param.sparse_plan.res_dims() == 0) {
+        if (param.dense_plan.res_stride.empty()) {
+            return stash.create<DoubleValue>(0.0);
```
Comment (arnej27959) [inline eval/src/vespa/eval/instruction/universal_dot_product.cpp:85]: could be wrong with trivial dimensions

File: eval/src/tests/instruction/universal_dot_product/universal_dot_product_test.cpp [hunk 2]
```diff
@@ -257,6 +294,39 @@ TEST(UniversalDotProductTest, generic_dot_product_works_for_various_cases) {
...
+    verify("reduce(x1_7y8z2*x1_1y8z2,sum,y)");
+}
+
+TEST(UniversalDotProductTest, forwarding_expanding_reduce) {
+    verify("reduce(5.0*y0_0,sum,y)");
+    verify("reduce(z16*y0_0,sum,y)");
+    verify("reduce(x1_1*y0_0,sum,y)");
+    verify("reduce(x0_0*y1_1,sum,y)");
+    verify("reduce(x1_1z16*y0_0,sum,y)");
+    verify("reduce(x0_0z16*y1_1,sum,y)");
```
Comment (arnej27959) [inline eval/src/tests/instruction/universal_dot_product/universal_dot_product_test.cpp:316]: consider
`    verify("reduce(x1z8*y0_0z8,sum,y,z)");
`

**Review by arnej27959 — DISMISSED** [review]
looks good with one minor suggestion

**Comment by havardpe** [comment]
@arnej27959 PTAL

**Review by arnej27959 — APPROVED** [review]
👍
